opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java

```diff
@@ -0,0 +1,243 @@
+import io.opencensus.trace.config.TraceConfig;
+import io.opencensus.trace.config.TraceParams;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+public class OpenTelemetrySpanBuilderImpl extends SpanBuilder {
+  private static final Tracer TRACER =
+      OpenTelemetry.getGlobalTracer("io.opencensus.opentelemetry.migration");
+  private static final Tracestate TRACESTATE_DEFAULT = Tracestate.builder().build();
+  private static final TraceOptions SAMPLED_TRACE_OPTIONS =
+      TraceOptions.builder().setIsSampled(true).build();
+  private static final TraceOptions NOT_SAMPLED_TRACE_OPTIONS =
+      TraceOptions.builder().setIsSampled(false).build();
+
+  private final String name;
+  private final Options options;
+
+  private List<Span> parentLinks = Collections.emptyList();
+  private final List<io.opentelemetry.api.trace.SpanContext> otelParentLinks = new ArrayList<>();
+  @Nullable private final Span parent;
+  @Nullable private final SpanContext remoteParentSpanContext;
+  @Nullable private Sampler sampler;
+  @Nullable private Boolean recordEvents;
+  @Nullable private io.opentelemetry.api.trace.Span.Kind kind;
+
+  @Override
+  public SpanBuilder setSampler(Sampler sampler) {
+    this.sampler = checkNotNull(sampler, "sampler");
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setParentLinks(List<Span> parentLinks) {
+    this.parentLinks = checkNotNull(parentLinks, "parentLinks");
+    for (Span parent : parentLinks) {
+      this.otelParentLinks.add(
+          io.opentelemetry.api.trace.SpanContext.create(
+              io.opentelemetry.api.trace.TraceId.bytesToHex(
+                  parent.getContext().getTraceId().getBytes()),
+              io.opentelemetry.api.trace.SpanId.bytesToHex(
+                  parent.getContext().getSpanId().getBytes()),
+              TraceFlags.getDefault(),
+              TraceState.getDefault()));
+    }
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setRecordEvents(boolean recordEvents) {
+    this.recordEvents = recordEvents;
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setSpanKind(@Nullable Kind kind) {
+    this.kind = mapKind(kind);
+    return this;
+  }
+
+  @Override
+  public Span startSpan() {
+    // To determine whether to sample this span
+    TraceParams activeTraceParams = options.traceConfig.getActiveTraceParams();
+    Random random = options.randomHandler.current();
+    TraceId traceId;
+    SpanId spanId = SpanId.generateRandomId(random);
+    Tracestate tracestate = TRACESTATE_DEFAULT;
+    SpanContext parentContext = null;
+    Boolean hasRemoteParent = null;
```

Comment:
```suggestion
    boolean hasRemoteParent = false;
```

Reply:
I think

Reply:
I think the null is necessary here since the sampler's comment says

> @param hasRemoteParent {@code true} if the parent {@code Span} is remote. {@code null} if this is a root span.